### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by mateuszkocz**

### CODE REVIEW

File: src/components/Authorization.js [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+            }
+        }))
+    }
+
+    unsetCredentials() {
+        this.setCredentials({token: null, name: null, id: null})
+    }
+
+    isUserValid({name, id} = {}) {
+        return Boolean(name && id)
```
Comment (Tuhaj) [inline src/components/Authorization.js]: I could be more certain with using: `Boolean(name) && Boolean(id)`

File: src/components/Givers.js [hunk 2]
```diff
@@ -65,15 +47,19 @@ class Givers extends Component {
...
+    render() {
         const {year, month} = this.props.match.params
         return (
-            <div>
-                <DateNavigation currentYear={year} currentMonth={month}/>
-                {
-                    this.state.loading ? <Spinner/> : this.renderStatsSection()
-                }
-            </div>
+            <>
```
Comment (Tuhaj) [inline src/components/Givers.js]: so this is equivalent to `React.Fragment`?

File: src/components/Header.js [hunk 3]
```diff
@@ -20,21 +21,31 @@ class Header extends Component {
...
+                                    </NavLink>
+                                </li>
+                                <li>
+                                    <NavLink to={"/stats"} activeStyle={{color: Colors.Banana}} exact>
+                                        Statistics
+                                    </NavLink>
+                                </li>
+                                <li>
+                                    <NavLink to={`/stats/givers/${year}/${month}`} activeStyle={{color: Colors.Banana}}>
+                                        TEMP: Givers
```
Comment (Tuhaj) [inline src/components/Header.js]: I think it's not temp, but permanent ;-) It's a comment, lets write just Givers

File: src/components/KudosRequest.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React, { Component } from 'react'
+import Request, { RequestMaker } from './Request'
+
+export default class KudosRequest extends Component {
+    static propTypes = RequestMaker.handlersPropTypes
+    static defaultProps = RequestMaker.defaultProps
+
+    normalize = kudos =>
+        kudos.reverse()
```
Comment (Tuhaj) [inline src/components/KudosRequest.js]: oh, you left this method, is it really alright?

File: src/components/Authorization.js [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+const AuthorizationContext = createContext(null)
+
+class Authorization extends Component {
+    // Store key names.
+    static tokenStorageName = 'token'
+    static userStorageName = 'user'
+
+    // Query params passed from the API during login.
+    static authTokenParamName = 'token'
```
Comment (Tuhaj) [inline src/components/Authorization.js]: again token? Not sure if correct when in line 9 is also 'token'

File: src/components/LoginScreen.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+class LoginScreen extends Component {
+    render () {
+        return (
+            <Container>
+                <WelcomeText>Kudos for coming!</WelcomeText>
+                <IntroText>
+                  This app helps to show our thankfulness.
+                </IntroText>
+                <LoginWithSlackButton/>
+                <Description>We&#39;re happy you're here! <span role="img" aria-label="Hug emoji">🤗</span></Description>
```
Comment (Tuhaj) [inline src/components/LoginScreen.js]: this second quote breaks here formating on github :(

File: src/components/LoginWithSlackButton.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+const redirectUri = encodeURIComponent('https://kudosapp-staging.herokuapp.com/slack/redirect')
+const slackUrl = `https://slack.com/oauth/authorize?scope=${scope}&client_id=${clientId}&redirect_uri=${redirectUri}`
+
+class LoginWithSlackButton extends Component {
+    render () {
+        return (
+            <a href={slackUrl}>
+                <img
+                    alt="Login with Slack"
+                    src="https://platform.slack-edge.com/img/sign_in_with_slack.png"
```
Comment (Tuhaj) [inline src/components/LoginWithSlackButton.js]: I suggest to extract the `SLACK_EDGE_BASE_URL`

File: src/components/Request.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+        ...Request.handlersPropTypes
+    }
+
+    static defaultProps = {
+        then: () => undefined,
+        catch: () => undefined
+    }
+
+    static get endpoint() {
+        return 'https://kudosapp-staging.herokuapp.com/'
```
Comment (Tuhaj) [inline src/components/Request.js]: Can we have an env for it?

File: src/components/Authorization.js (hunk 1 above)
Comment (mateuszkocz) [reply src/components/Authorization.js]: Certain of what?

File: src/components/Givers.js (hunk 2 above)
Comment (mateuszkocz) [reply src/components/Givers.js]: Yes. It's a shorter syntax.

File: src/components/KudosRequest.js (hunk 4 above)
Comment (mateuszkocz) [reply src/components/KudosRequest.js]: What do you mean? This behaviour is copied from its initial place in the Wall component. No business logic changed, AFAIK.

File: src/components/Authorization.js (hunk 5 above)
Comment (mateuszkocz) [reply src/components/Authorization.js]: Token in the 9th line refers to the local storage key. This one is a name of the param coming from backend. Their equality is coincidental.

File: src/components/LoginScreen.js (hunk 6 above)
Comment (mateuszkocz) [reply src/components/LoginScreen.js]: GitHub sucks. Let's more to exchanging .rar.

File: src/components/LoginWithSlackButton.js (hunk 7 above)
Comment (mateuszkocz) [reply src/components/LoginWithSlackButton.js]: It's nor really configurable. You can't have different values depending on an environment.

File: src/components/Request.js (hunk 8 above)
Comment (mateuszkocz) [reply src/components/Request.js]: Dunno. The URL was moved here as it was previously hard-coded in the (now removed) [KudosApi.js](https://github.com/Appnroll/kudosapp-frontend/blob/1e0ccb757173e228fdced7c0c9e2064183f3c0ff/src/api/KudosApi.js).

**Pull request merged**